ui/pages/swaps/view-quote/view-quote.js

```diff
@@ -672,93 +680,147 @@ export default function ViewQuote() {
-        })}
-      >
-        {selectQuotePopoverShown && (
-          <SelectQuotePopover
-            quoteDataRows={renderablePopoverData}
-            onClose={() => setSelectQuotePopoverShown(false)}
-            onSubmit={(aggId) => dispatch(swapsQuoteSelected(aggId))}
-            swapToSymbol={destinationTokenSymbol}
-            initialAggId={usedQuote.aggregator}
-            onQuoteDetailsIsOpened={quoteDetailsOpened}
-          />
-        )}
-
-        {showEditGasPopover && networkAndAccountSupports1559 && (
-          <EditGasPopover
-            transaction={{
-              userFeeLevel: swapsUserFeeLevel || GAS_RECOMMENDATIONS.HIGH,
-              txParams: {
-                maxFeePerGas,
-                maxPriorityFeePerGas,
-                gas: maxGasLimit,
-              },
-            }}
-            minimumGasLimit={usedGasLimit}
-            defaultEstimateToUse={GAS_RECOMMENDATIONS.HIGH}
-            mode={EDIT_GAS_MODES.SWAPS}
-            confirmButtonText={t('submit')}
-            onClose={onCloseEditGasPopover}
-          />
-        )}
-
-        <div
-          className={classnames('view-quote__warning-wrapper', {
-            'view-quote__warning-wrapper--thin': !isShowingWarning,
-          })}
-        >
-          {viewQuotePriceDifferenceComponent}
-          {(showInsufficientWarning || tokenBalanceUnavailable) && (
-            <ActionableMessage
-              message={actionableBalanceErrorMessage}
-              onClose={() => setWarningHidden(true)}
+    <GasFeeContextProvider
+      editGasMode={EDIT_GAS_MODES.SWAPS}
+      minimumGasLimit={usedGasLimit}
+      transaction={transaction}
+    >
+      <TransactionModalContextProvider captureEventEnabled={false}>
+        <div className="view-quote">
+          <div
+            className={classnames('view-quote__content', {
+              'view-quote__content_modal': disableSubmissionDueToPriceWarning,
+            })}
+          >
+            {selectQuotePopoverShown && (
+              <SelectQuotePopover
+                quoteDataRows={renderablePopoverData}
+                onClose={() => setSelectQuotePopoverShown(false)}
+                onSubmit={(aggId) => dispatch(swapsQuoteSelected(aggId))}
+                swapToSymbol={destinationTokenSymbol}
+                initialAggId={usedQuote.aggregator}
+                onQuoteDetailsIsOpened={quoteDetailsOpened}
+              />
+            )}
+
+            {!EIP_1559_V2_ENABLED &&
+              showEditGasPopover &&
+              networkAndAccountSupports1559 && (
+                <EditGasPopover
+                  transaction={transaction}
+                  minimumGasLimit={usedGasLimit}
+                  defaultEstimateToUse={GAS_RECOMMENDATIONS.HIGH}
+                  mode={EDIT_GAS_MODES.SWAPS}
+                  confirmButtonText={t('submit')}
+                  onClose={onCloseEditGasPopover}
+                />
+              )}
+            <EditGasFeePopover />
```

Comment:
Don't we want to render EIP 1559 v2 components only if `EIP_1559_V2_ENABLED` is true? Because otherwise they would slow down rendering a little.

Reply:
Good point, I made this change.